src/main/java/ro/societateahermes/backendservice/service/UserServiceInterface.java

```diff
@@ -1,11 +1,19 @@
 package ro.societateahermes.backendservice.service;
 
+import ro.societateahermes.backendservice.entities.DTO.MySubmissionDTO;
+import ro.societateahermes.backendservice.entities.DTO.UserDTO;
+import ro.societateahermes.backendservice.entities.Participation;
 import ro.societateahermes.backendservice.entities.User;
 
 import java.util.List;
 
 public interface UserServiceInterface {
+
     void save(User user);
 
+    User saveUserFromDTO(MySubmissionDTO user);
+
+    void addParticipation(User user, Participation participation);
```

Comment:
Ar trebui sa primeasca de la controller un userDto

Reply:
In controller am partea asta
public void submit(@RequestBody @Valid MySubmissionDTO submission) {
        submissionService.savefromDTO(submission);
        User user = userService.saveUserFromDTO(submission);
        Participation participation = participationService.savefromDTO(user, submission);
        userService.addParticipation(user, participation);
        eventService.addParticipation(submission.getEventId(), participation);
}
Primeste un user si participation, deoarece pe baza dto ului primit de la frontend acestea se salveaza in bd, la salvare returnandu-se si obiectul pentru a avea id ul......aici nu vad necesar crearea unui dto,daca gresesc sa imi spui

Reply:
Daca faci asa atunci nu o sa ai in baza de date acelasi user de doua ori?

Reply:
user ul si eventul au o lista de participation pentru a putea face maparea intre tabele, atunci cand se submite formularul se salveaza userul, participarea e tabela de legatura intre user si event si atunci participation are o referinta atat catre user cat si catre event......atunci cand primeste user si participation, doar adauga in lista din user participation ul primit ( metoda asta am pus-o initial in user si ai zis sa o mut in service)